web/frontend/src/pages/ballot/components/Rank.tsx

```diff
@@ -102,13 +109,25 @@ const Rank: FC<RankProps> = ({ rank, answers, language }) => {
               <ul className={rank.ID} {...provided.droppableProps} ref={provided.innerRef}>
                 {Array.from(answers.RankAnswers.get(rank.ID).entries())
                   .map(([rankIndex, choiceIndex]) => {
-                    if (language === 'fr' && rank.ChoicesMap.has('fr'))
-                      return choiceDisplay(rank.ChoicesMap.get('fr')[choiceIndex], rankIndex);
-                    else if (language === 'de' && rank.ChoicesMap.has('de'))
-                      return choiceDisplay(rank.ChoicesMap.get('de')[choiceIndex], rankIndex);
-                    else if (rank.ChoicesMap.has('en'))
+                    if (language === 'fr' && rank.ChoicesMap.ChoicesMap.has('fr'))
+                      return choiceDisplay(
+                        rank.ChoicesMap.ChoicesMap.get('fr')[choiceIndex],
+                        rank.ChoicesMap.URLs[choiceIndex],
+                        rankIndex
+                      );
+                    else if (language === 'de' && rank.ChoicesMap.ChoicesMap.has('de'))
+                      return choiceDisplay(
+                        rank.ChoicesMap.ChoicesMap.get('de')[choiceIndex],
+                        rank.ChoicesMap.URLs[choiceIndex],
+                        rankIndex
+                      );
+                    else if (rank.ChoicesMap.ChoicesMap.has('en'))
```

Comment:
```suggestion
                    if (rank.ChoicesMap.ChoicesMap.has(language))
                      return choiceDisplay(
                        rank.ChoicesMap.ChoicesMap.get(language)[choiceIndex],
                        rank.ChoicesMap.URLs[choiceIndex],
                        rankIndex
                      );
                    else if (rank.ChoicesMap.ChoicesMap.has('en'))
                      return choiceDisplay(
                        rank.ChoicesMap.ChoicesMap.get('en')[choiceIndex],
                        rank.ChoicesMap.URLs[choiceIndex],
                        rankIndex
                      );
```

which would get rid of the third `if`.